### DISCUSSION TITLE
MongoUpgrade Issue:Spring Mongo

### DISCUSSION THREAD

**Discussion by cijugeorgie** [flapdoodle-oss/de.flapdoodle.embed.mongo#516] [Q&A] [answered]
We are upgrading the mongo db. While running test cases in the pipeline, we are getting this error. Can anyone please advise. Locally(Window) its runs good.

Error creating bean with name 'embeddedMongoServer' defined in org.springframework.boot.autoconfigure.mongo.embedded.EmbeddedMongoAutoConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [de.flapdoodle.embed.mongo.MongodExecutable]: Factory method 'embeddedMongoServer' threw exception; nested exception is java.lang.IllegalArgumentException: could not resolve package for Version{4.4.16}:Platform{operatingSystem=Linux, architecture=X86_64

I have this in build.gradle

testImplementation 'de.flapdoodle.embed:de.flapdoodle.embed.mongo.spring27x:4.6.2' and   mongodb.embedded.version: = 4.4.16 in application.yml

**Comment by quxelteak** [comment]
This may be because the flapdoodle library does not support the version of mongo db you are using. You could try upgrading flapdoodle

**Comment by michaelmosmann** [comment]
@cijugeorgie what kind of linux is running in the ci pipeline?

**Comment by cijugeorgie** [comment]
Hi Mossman..it's Debian.

Error message also says that

**Comment by cijugeorgie** [comment]
Thanks for the response Nevin. I tried different versions including spring
30. It's not working. Which one to use is confusing as there are multiple
versions

**Comment by michaelmosmann** [comment]
@cijugeorgie .. from the small part of your error message i guess that your debian version is not detected ( Platform{operatingSystem=Linux, architecture=X86_64[missing part?]}) or this mongodb version does not exist for your platform.. 

can you provide a more complete stack trace/error message?

spring versions: which spring versions do you use? (spring boot, spring something?)

**Comment by michaelmosmann** [comment]
@cijugeorgie 
the latest spring 2.7.x integration is 4.16.6 (https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo.spring/tree/spring-2.7.x)

**Comment by cijugeorgie** [comment, marked as answer]
Thanks for the response Mike.

We were using 2.7.18 Spring version.

We were trying out different versions to fix and somehow when we changed it
to mongodb.embeded.version to 4.0.2 in application.yml, it started working
fine
